hack/check-license-headers.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+#
+# SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
```

Comment:
This is very similar to `api/hack/check-generate.sh` - can we extract the common parts as reusable functions which can then be included wherever required?